src/test/java/seedu/address/model/tag/MedicationMapTest.java

```diff
@@ -0,0 +1,88 @@
+package seedu.address.model.tag;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class MedicationMapTest {
+
+    private final MedicationMap medicationMap;
+
+    public MedicationMapTest() {
+        medicationMap = new MedicationMap();
+        medicationMap.add(new Medication("Paracetamol"));
+        medicationMap.add(new Medication("Ibuprofen"));
+    }
+
+    @Test
+    public void add_nullMedication_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> medicationMap.add(null));
+    }
+
+    @Test
+    public void add_validMedication_success() {
+        Medication medication = new Medication("Aspirin");
+        medicationMap.add(medication);
+        assertEquals(medicationMap.size(), 3);
+        // clean up
+        medicationMap.remove(medication);
+    }
+
+    @Test
+    public void remove_medicationNotInMap_success() {
+        Medication medication = new Medication("Aspirin");
+        medicationMap.remove(medication);
+        assertEquals(medicationMap.size(), 2);
+    }
+
+    @Test
+    public void remove_medicationInMap_success() {
+        Medication medication = new Medication("Paracetamol");
+        medicationMap.remove(medication);
+        assertEquals(medicationMap.size(), 1);
+        // clean up
+        medicationMap.add(medication);
+    }
+
+    @Test
+    public void setStringifiedMedicationMap_validString_success() {
+        String medicationMapString = "Paracetamol /-/-/ 1 /-/-/ Ibuprofen /-/-/ 1";
+        MedicationMap medicationMap = new MedicationMap();
+        medicationMap.setStringifiedMedicationMap(medicationMapString);
+        assertEquals(medicationMap.size(), 2);
+    }
+
+    @Test
+    public void setStringifiedMedicationMap_emptyString_success() {
+        String medicationMapString = "";
+        MedicationMap medicationMap = new MedicationMap();
+        medicationMap.setStringifiedMedicationMap(medicationMapString);
+        assertEquals(medicationMap.size(), 0);
+    }
+
+    @Test
+    public void toString_validMedicationMap_returnsString() {
+        assertEquals("\nTypes of medication: 2\n"
```

Comment:
Perhaps it's better to abstract out the expected toString into a variable called `expectedMessage`. This prevents confusion of all the String concatenation within the `assertEquals` method.
This applies to the following few tests within this class as well.